code/_helpers/turfs.dm

```diff
@@ -144,7 +144,8 @@
 	for(var/obj/O in T)
 		if(O.simulated)
 			O.loc = X
-			O.update_light()
+			if(O.light_system == STATIC_LIGHT)
+				O.update_light()
```

Comment:
In this case since it sets loc, the movable light might not come with, so you might want to do like...

```suggestion
			if(O.light_system == STATIC_LIGHT)
				O.update_light()
			else
				var/datum/component/overlay_lighting/OL = O.GetComponent(/datum/component/overlay_lighting)
				OL?.on_parent_moved(O, X, O.dir, TRUE)
```

I thiiiiink that will make sure it comes with?